### ISSUE TITLE
Is Semantic UI development dead ?

### ISSUE THREAD

**Issue by GuacheSuede** [Semantic-Org/Semantic-UI#6109]
@jlukic  's last commit was 5/6  months ago and even then did not have many improvements nor much changes.

What would be the status of Semantic UI especially with Bootstrap having much more development than this ? Is it in active development or are any features planed for the near future ?

Note: This open source project goes dead without @jlukic or @quirkyjack , zero developement, zero updates, and for nearly a year at that. Even pull requests are not moving. Could there perhaps be a solution to this ?

Please do not say plans for something, for that might be for another half a year of waiting with no promises or answers

There will be a meeting at 6PM EST tomorrow on Hangouts !

**I hope to volunteer to maintain/run this project if @jlukic does not have the time for it**

**Comment by arjarn** [comment]
Many many people are same interrogation : Semantic UI is great framework (with default but lot of good points). I'm really sad to see development stopped. For development (newer), using a framework without news about futur is not attractive.

**Comment by etshy** [comment]
Same here I'd love to see semantic continue its development.
Maybe the semantic team works on something huge, like a V3 ?

**Comment by GuacheSuede** [comment]
@etshy not in the Semantic-Org Repo, and @jlukic activity has been low too.

**Comment by jeveloper** [comment]
Nothing since august last year

I've decided to move to material ui, this isnt worth investing time

I would however be happy to pay an annual fee for a premium plan that will guarantee bug fixes and improvements.

**Comment by GuacheSuede** [comment]
@jeveloper I agree. perhaps a fork could be created

**Comment by arjarn** [comment]
Agree too. I'm developper and do little stuff for fun. This month, I  begin new project (not for work) and have to choose between SUI and Metro-UI CSS V4 (I'm already using V3). Finally I choose Metro-UI : lot of high level components and very active developper.

Too bad : I like SUI, it's very good framework but no dev and with time, chatroom become empty... so no support.

For fork, possible if people are ready to participate...

**Comment by GuacheSuede** [comment]
@arjarn Do you happen to know what is needed for Semantic UI to be better ?

**Comment by arjarn** [comment]
In my opinion, SUI is almost perfect ^^
But :

  - be more easily customisable (skin) : actually it's difficult to modify color or add background on component (or I'm bad... possible because newer with SUI  ;-) )
  - several components missing (need 3rd party now) : date / time picker or have image slideshow.

But it's my feeling after my poor SUI experience, why don't making "BOX IDEA" here or elsewhere ?

PS : sorry for my poor english.

**Comment by hammy2899** [comment]
I have been considering making a fork for a while now and I would love to help get peoples features added and patch the current bugs.

I have some knowledge for the semantic code base but I will have to learn some more before I make a fork.

I will 100% make a fork if people are willing to contribute. 😃

**Comment by GuacheSuede** [comment]
@hammy2899 Count me in, I think it's doable.

**Comment by witrin** [comment]
@GuacheSuede @hammy2899 @arjarn @jeveloper @etshy Please read the following:
 * https://gitter.im/Semantic-Org/Semantic-UI?at=5a3ca96fffa3e37919693fde
 * https://gitter.im/Semantic-Org/Semantic-UI?at=5a3caa595355812e5705bfe2
 * https://gitter.im/Semantic-Org/Semantic-UI?at=5a3cabadc072deaf0bee5281
 * https://gitter.im/Semantic-Org/Semantic-UI?at=5a3cad03ffa3e379196950bc
 * https://gitter.im/Semantic-Org/Semantic-UI?at=5a501e8c03838b2f2a7760ae

**Comment by hammy2899** [comment]
@witrin thanks for bringing this up!

**Comment by Bodom78** [comment]
@witrin Good to know. 

Hopefully @jlukic finds a way to finance future development as this framework has some great concepts.

**Comment by GuacheSuede** [comment]
@witrin I agree with his sentiments, but what seems to be an issue is that this open source project goes dead without @jlukic, zero developement, zero updates, and for nearly a year at that. Even pull requests are not moving. Could there perhaps be a solution to this ?

**Comment by GuacheSuede** [comment]
@Bodom78 Could you share what great concepts Semantic UI has from your POV ? For me it is mostly Modular Theme-ing

**Comment by philrykoff** [comment]
As there seem to be people willing to support SUI, maybe @jlukic considers adding them to the team.

**Comment by arjarn** [comment]
@witrin : thx for information

+1 @Bodom78  : like SUI for same things, can add bricks to make great site with original design, but I haven't find easy way for theming so for this, I'm agree @GuacheSuede 

And, obviously, priority to resolve issues.

In last, more high level components : I think, dev / people like have all-in-one framework. Tiresome have 4-5 third party library for image slide, date time picker or other...
I know, 100% needing is not possible but several components missing in SUI in my mind.

**Comment by GuacheSuede** [comment]
That is good to hear ! @Bodom78 @arjarn 

@Bodom78  I agree, SUI's feature of markup seems much easier than Boostrap or Foundation. I agree that the variations of the the framework seems wonderful !

@arjarn Thanks for agreeing :). I agree SUI is lacking a few features that would be much desired.

Issues are a large priority.

I am thinking it would be better for Semantic UI if it was switched to SASS/SCSS due to more features

**Comment by GuacheSuede** [comment]
I have so far not been able to get @jlukic reponse when I private messaged him a few days ago

**Comment by Joshfindit** [comment]
Since the discussion includes forking, I suggest that future maintainers seriously consider moving Semantic-UI to CSS Grid.

**Comment by GuacheSuede** [comment]
@Joshfindit Definitely would, do you have an example of a framework that does this ?

**Comment by witrin** [comment]
Here are my two cent:

For me it's the core concept of this Framework what makes it so valuable and (of course) the rich feature set.

We are using this framework with our custom build tool chain (YARN only with Fluid templates) and the big effort with this concept is separation of concern! We don't need to copy fat boiler plates for each project, we reuse our investments, we spent less time and the projects are just that big they only have to be. So the possibility to layering components with definitions, themes and sites are a huge step forward in this (CSS) area.

Also having a clean wording makes it more fun to work with this stuff ... but I know structures needs to grow and kept clean, so the only thing I'm most aware of when a fork would came up, core concepts and/or structure became a mess because too many cooks (might) spoil the broth. E.g. when tickets like might accepted in a hurry https://github.com/Semantic-Org/Semantic-UI/issues/5785.

I also think the @jlukic should be still part of this; sure, if he want. Maybe he could find a way the community could be involved a little bit more, thus we get faster releases and he gets more free time. I also see financial support for this project should be a part of the future discussions.

**Comment by GuacheSuede** [comment]
@witrin Yes definitely. 

I agree too many cooks (might) spoil the broth and that @jlukic should be part of this as well as financial support as an issue to be brought up.

What I am suggesting however would be a change of operation for the project has come to a halt for a year and it would be a waste if it died especially as you said of the good concepts of it (modularity, build tool chain, rich feature set and wording)

I assure you that I totally agree with you especially on the key point of keeping structure and maintaining quality.
But the dilemma is of such that Semantic UI is not moving with @jlukic nearly disappearing, remember this is a 32 thousand stars project.
Do you think a Semantic UI should stay dead ?

**Comment by witrin** [comment]
@GuacheSuede Semantic UI should definitely kept alive! :D I think the next step could be a discussion about a change of operation for the project; like you mentioned. If I get the last messages from @jlukic right, I think he is currently in Russia on vacation. So maybe we wait a little until he pings back and a discussion about operation then could be forced; and we will see or hope for the best? ^^ And when he really would not answer within the next `n` weeks (what I don't think) maybe a discussion about a community driven fork needs to be done earlier.

**Comment by GuacheSuede** [comment]
@witrin Yes, I agree. Glad we are on the same page and understanding, perhaps let's wait till the Mid Feb  and see if he responds till then ?

**Comment by levithomason** [comment]
Note, Semantic UI React is still in development.  I'm the owner/author over there and I have owner rights to the Semantic Org at large.  I've also had offline chats with Jack about the future of SUI.

Jack's focus is mostly on pursuing a semantic component standard for the web at large.  A W3C for UI, if you will.  Frameworks like Semantic UI and library implementations like Semantic UI React would serve as premier examples of how to implement the UI specification.

All said, I am interested in keeping development moving forward on both Semantic UI and Semantic UI React.  The next frontier, in my mind, is to abstract out the styles and state of the components so that other implementations can more easily adapt them.  It would also serve to protect us against future changes in the technology landscape.  I've discussed this with Jack on a video call and he is in agreement, regarding the abstracted styles at least.

In my experience here at Semantic Org, what we need is a core of truly dedicated members.  Over the last few years, I've seen a lot of conversations like this one begin, and sometimes an effort is started, but there is no follow through.  I think the reasons behind the lack of follow-through is what we need to address.

One idea I've had is to start with regular meetings.  At least get a core group of folks together to talk about a realistic way to move forward.  I am more than happy to hear thoughts as to how we can get folks to commit in a sustainable and productive manner.

Who's up for it?

**Comment by littledannytables** [comment]
I have been in contact with @jlukic for the past couple of weeks and hopefully there will be a patch before the 1st of Februari. He has been quite busy lately.

**Comment by dannyBies** [comment]
I have been in contact with @jlukic for the past couple of weeks and hopefully there will be a patch before the 1st of Februari. He has been quite busy lately.

**Comment by jlukic** [comment]
I think it'd be reasonable to have a video call with people who are interested in SUI to discuss what's going on in the project. 

I'll summarize the story of the project here, as I know I've told it through different mediums at different times and it's probably more difficult to track these days. 

My goal is to share my thoughts as I understand them myself, naked without a filter, and I hope you can do me the kindness of being forgiving for any parts where I may have faltered in character or conclusion.

I've generally thought of the core SUI goals as a [descriptivist standard](https://harpers.org/wp-content/uploads/HarpersMagazine-2001-04-0070913.pdf) for web UI, a counterpoint to the top-down approach of the W3C. This is something that is still my primary life ambition, and that I hope I can be thinking about until my later days.  The issue though, is that indefinitely financing open source can be difficult. It, at the very least, requires enough financing to cover one persons expenses indefinitely which, in real financial terms, is a multi-million dollar affair. I'll get to that in a bit though. Perhaps first I'll try to share the story of where the project came and where I am now.

---------

Semantic UI 1.0 was lucky enough to exist because I was able to open source much of my work done at the [now defunct startup Quirky](http://fortune.com/2015/09/22/quirky-files-bankruptcy/) where I was working as head of front end development on solving the general issue of how large teams can work on a front end codebase without developer scaling issues. The aspirations for the project were to show how aspects of semantics from natural language, (noun modifier relationships, plurality, significant word order) were better at describing physicality than classic CS models which work more like 'cooking recipes' (or behavioral specs). 

When I left Quirky I ended up travelling a bit and working on Semantic UI, generally spending through savings and hammering out ideas around types of UI, naming conventions, reasonable methodologies for constructing UI definitions and generally looking for a path forward. Around a year into this plan I got an interesting email from someone who had recently bought [The New Republic](https://newrepublic.com/). I ended up coming in and discussing what I was doing, and found out it was a good match. I was offered an interesting arrangement where I could work to help recode the magazines aging codebase (using SUI) and redesign the magazines website, on a 75/25 basis while working on SUI one week a month without pay out of their office.

After around 6 months the redesign was done, and I was itching to get back to SUI full-time. Chris was kind enough to offer me the opportunity to keep a desk at the New Republic offices while working on open source full-time. 

I spent the next year, keeping a desk at their Union Square office, greeting guests as if I was an employee, but working without pay on open source full-time, from summer  2014 through summer 2015, when i finally finished 2.0 of Semantic UI.

This was my second period working entirely from savings and taking no salary while working on SUI, and it was extremely liberating and stimulating. In all honesty, it was perhaps the best and most fulfilling period of my life. Though at the end of it, on medicaid with no money, I knew it wasn't going to sustainable. I had now saved and then spent ~$60-80,000 in savings in two separate periods working on Semantic UI.

At this time I started talking to some investors around perhaps raising a seed round for a company supporting Semantic UI. From talking to investors and angels, I was introduced to some interesting people who founded other OS companies like [Meteor](https://www.meteor.com/), and [Ionic](https://ionicframework.com/) and had calls to discuss their story.  I also started talking to a list of valley VCs, preparing my seed round slide deck and getting ready to sell this ideas. 

But something hit me in my gut, that felt wrong about it. When I think about the people that have inspired me to create in my life, the idea of attaching a 'profit motive' created cognitive dissonance. There was no "Dostoevsky LLC", or "James Joyce C Corp". Perhaps, one could argue, there were publishers, but these arrangement didn't require the same quick turnarounds and preposterous hockey puck revenue charts as is expected in the valley these days.

The idea of attaching a company, and shareholders to SUI didn't feel right to me, and from looking through the direction of open source companies who had tried, it did seem clear all of them, without falter ended up following the money eventually. At its worst, if I relented and tried anyway and decided it wasn't correct the intellectual property for SUI would, most likely, never be available again--owned by the stakeholders in the company.

While I was weighing these thoughts, a few SUI companies were reaching out about 2.0 and how to use SUI in their company. One of those companies [Qualia](http://www.qualia.com) was starting to raise their series A and asked me to come out and look at their codebase and give feedback on their design. They were fresh Stanford and Penn grads using an entirely Meteor/SUI stack, and with very stimulating views on code and business. From talking with them, it was clear I would be able to 1) have access to a significant equity stake in an early stage company that had not been available in any other period of my life 2) be able to use SUI as the core of a startups codebase and test out longitudinally the utility and issues. 

Around fall 2015 I ended up signing contracts and starting full time as head of design at Qualia, something that I was very sheepish to talk about publicly because it was a difficult conceit to explain. But I was penned in for two years, which seemed in the scope of a lifelong project as "not forever", and so I soldiered on. 

I ended up agreeing to a "week on week off" schedule from NYC -> SF. Which in hindsight, may have been a very silly thing, leaving very little weekend time for open source that wasn't recuperating or traveling, or reconnecting with friends. My 2 years ran up in July of this year, and I was offered the same equity to stick another 2 years, a contract I accepted.

Now it's not difficult to see how this could be construed as 'selling out', but having gone through my life savings twice, having tried the contracting/OS balance, having looked at SUI as an excel spreadsheet of a seed deck's financial projections, and having calculated the multiple of current donations that i'd need to receive to be able to pay rent, it didnt seem feasible to go any other way.

----------------

The only way forward I can imagine for the project is to raise, at once, enough money to carry the product indefinitely, what I've been thinking of as 'escape velocity' for OS. This number begins at the cost of one developer to work, in perpetuity, on an OS project. The calculations for this looks something like this, the numbers may vary somewhat, but coarsely still look the same.

[Cost of living NYC](http://www.businessinsider.com/salary-to-live-in-new-york-city-2015-10)
$85,000

Cost of living with savings and possibility of family
$125,000

Safe rate of return for annuity/investment
4%

Endowment/Annuity that could produce cost of living indefinitely
$125,000 / 0.04% = $3,125,000

Long term capital gains tax 
20%

Amount of pre-tax money necessary to finance an endowment for one-man Semantic UI non profit:
$3.906,000

-----

All and all this is a fair amount of money that doesn't seem reasonable to raise without institutional investment, or personally putting up enough capital to cover the lion's share of the non-profit. 

I'm trying my best to do the latter for now, which has caused me to make some clear and perhaps single-minded concessions for the project. 

All and all though, I think if I'm capable of financing a non-profit, where I can say *for certain* my children will be able to have long term access to an open source UI framework (or standard) without profit motive, or need for solicitation, I'll be very proud of this time.

**Comment by Joshfindit** [comment]
@GuacheSuede: Currently there is no framework that I'm aware of that is based on CSS Grid. It's very new (but has excellent browser support), and I think this is a technical opportunity as well as an SEO opportunity: CSS Grid does a lot of things very intelligently and so obsoletes some of the old methods of page placement, and for as many people who talk about 'with CSS Grid we don't need [Bootstrap] any more', there are possibly more who recognize that CSS Grid doesn't bring design cohesion, design language, or even a design direction and these are really where Semantic-UI shines.

**Comment by Bodom78** [comment]
@jlukic Thanks for the update, it's good to know and have put into perspective the origins, difficulties and future wishes for the project. 

@levithomason If the project was to move towards adding some dedicated contributers to assist in maintaining the project, then I feel a "call to arms" would best be placed on the semantic website somewhere. I generally find Git issue threads is mostly users looking for fixes and messaging can get lost easily. (Looking forward to using Semantic UI React in the future!)

**Comment by jlukic** [comment]
If you'd like to participate in a conference call about Semantic UI, I've created this link for listing availability.

https://doodle.com/poll/2uqrrzbazigip4xg

**Comment by GuacheSuede** [comment]
@jlukic Thanks Jack for your long and candid explanations in relation to funding and your struggles.

I think that we need to talk yes and personally would like to manage the project if you do not happen to have time for this as i do see many new features that can be done that would benefit users.

I have filled the poll and look forward to talking to you soon.

**Comment by w96k** [comment]
@jlukic It's nice to hear any updates about SUI because even without actively maintaining SUI is the best OS CSS framework available. I think it would be nice to have more maintainers to review all pull requests and issues. Maybe you should look to React SUI, they tried to solve the problem of styled components, but without a result.

**Comment by jlukic** [comment]
For what it's worth i jammed out a release tonight
https://github.com/Semantic-Org/Semantic-UI/releases/tag/2.2.14

**Comment by arjarn** [comment]
Nice to see motivated people for this great project.
Agree with @jlukic's thought about SUI.

Unfortunately I haven't lot of time to devote for new project.
But I'm coming here for proposing ideas or issues ;)

Good luck for conference call, I cross my fingers for positive issue :)

**Comment by kurbar** [comment]
@jlukic I would be happy to help out as well. Unfortunately I wouldn't be able to join on those meeting times. Simply because I live in Eastern Europe and for me, those times are from 1am to 5am.

**Comment by brunotourinho** [comment]
Hey folks!
Been using Semantic-UI as my projects' framework for a long time.
It's good to hear that is not dead.
Thanks @jlukic for your honesty and of course for sharing your time and work with all of us.
Hope that you guys can manage to keep things going.

**Comment by hammy2899** [comment]
@jlukic It's great to hear what you're trying to do with the project and I fully support that and would love to see semantic grow into a standard for modern web. 

Over the past few weeks I have been contributing and helping out by updating some things and I want to help out as must as I can because I love this project and I have been using the framework for well over a year now, with that said I am hoping to become a contributor/maintainer so I can help sort out the issue right now with the current amount of issues/pr there is and get semantic in a stable bug free version with components for every scenario while keeping the style and easability which Jack is going for.

I have joined 2 of the days on the meeting planner which @jlukic posted so I can see what the next moves are and so I can help out in anyway I can.

**Comment by khornberg** [comment]
While I cannot make the discussion, I'll offer my thoughts here beforehand.

A community seems to be here to help. How do we capitalize on that desire for the health of the project?

* Open communication - this is critical for enlisting the help of others and "marketing" SUI for both funding and community help
* Small, focused tasks - these can be done for "free" when distributed to many contributors (this is more possible with tests; e.g. visual regression tests)
* Management and direction - these become a primary task of either the financially supported leader and/or other leaders of the project

Things I would like to see in the future:

* Making SUI Javascript framework/library agnostic - I recognize work is already going on to do this
* Incorporate features (e.g. calendar module) proactively - the community has already written them
* Regular releases - technology changes (e.g. CSS grid, drop IE support, etc.), bug fixes, merged features

I think these things are possible, albeit, slowly, regardless of funding if there is by-in by the current leaders of SUI.

**Comment by arjarn** [comment]
@khornberg : nice analysis. 

I cannot make the discussion too (schedule problem), so also offer my thoughts here...

Calendar and slider module are already written (for slider, I think it can be improving). 
Be great to incorporate it in SUI.
For slider, technically, why not same as Vegas Slide (https://vegas.jaysalvat.com/) : using background image in HTML component with optionals controls to swap between pictures. With that, it's possible to place above picture text, button... offering lot of possibilities as mostly SUI components ;)

Maybe others components can be adding : 

  - Toasts to display informative message (less perturbing user than modal message), like this http://materializecss.com/dialogs.html or https://metroui.org.ua/v4/toast.html 
  - Chips like http://materializecss.com/chips.html (like dropdown SUI)

and for design, wave effect when you activate button (like https://mdbootstrap.com/javascript/waves-effect)

All that is I'm dreaming in my devs : I know some ideas may be good, other very bad ^^

Last point, with multiples / heterogeneous device today, it may be useful to add two more breakpoints for responsive : one below 768px (576 ?) and one upper 1200px (1500 ?).

**Comment by hammy2899** [comment]
@arjarn I agree with getting the calendar and the slider (range) module merged into the core UI and I also agree with the idea to add toasts or some sort of notification yet I don't agree with the button ripple effect, this is a material design feature and in my option doesn't fit the design pattern for semantic. I think adding lots of features like this would make the framework too clunky when it is only required for that specific thing therefore you could just add it to your theme.

I definitely think a talk needs to be had about new and current components on what needs to be added and what needs to be changed to stay relevant to the current web standards and design.

**Comment by arjarn** [comment]
@hammy2899 I agree your opinion about button ripple. Perhaps not so good to mix components features and design feature.

**Comment by GuacheSuede** [comment]
@GammaGames Definitely sounds good !

**Comment by kappaknight** [comment]
What about a KickStarter or similar where some of the more desired components would be worked on as part of the reward? Font Awesome raised millions in the last year for version 5 of their product... maybe something like that could work for Semantic?

We just released the facelift of our new [automotive classifieds](https://www.easyautosales.com) website based completely on SUI. We definitely want to see this project continue moving forward. Once the site gets a bit more traction, we'd love to help out financially.

**Comment by dannyBies** [comment]
It seems 3, 6 and 7 Februari are the dates most people are available for a conference call about Semantic UI.
The most people can do the call on the 3rd, I will be out of country that day but I can make some time available to participate.

@jlukic are you organizing the call?

**Comment by DanAtShenTech** [comment]
Perhaps an endowment wouldn't be necessary.  ~40,000 stars  x  $5 / year.  I'd sure be willing to do that for a LONG time.

**Comment by italomaia** [comment]
What is the opinion of core devs on a kickstarter? font-awesome kickstarter seems to have done well for them.

**Comment by GammaGames** [comment]
If there was a kickstarter it would have to be [done well](https://blog.fontawesome.com/how-font-awesome-5-became-kickstarters-most-funded-software-project-4a63c035e84), but another alternative could be patreon. Just throwing some ideas out there for the people making decisions on the project

**Comment by menthol** [comment]
+1 for patreon,
@yyx990803 (Evan You) main developer of Vuejs framework, use Patreon to dynamise developpments of VueJs

**Comment by DanAtShenTech** [comment]
Perhaps a way to provide ongoing support would be to have an online SUI theme builder that would be paid (seems like most online theme builders are paid - or else the theme builder is free but you pay for the library).  I realize that this would require additional coding to put in place, but it could also be a tremendous help in spreading the use of Semantic UI.  A Kickstarter/IndieGoGo would be best for a project like that, of course.  A Patreon campaign could be started in the meantime if necessary.  I'm in for any of these ideas including simply long-term support through Patreon.

**Comment by GuacheSuede** [comment]
@DanAtShenTech I concur, will look into it. Depends if @jlukic has time, else i hope to maintain it

**Comment by jlukic** [comment]
I've closed out the poll for the conference call.

The date is set for [6PM EST](https://www.timeanddate.com/worldclock/converter.html?iso=20180202T230000&p1=179) tomorrow

I'll post the hangouts link in this thread around that time, so anyone who is around can join as well.

**Comment by Max-Zviagintsev** [comment]
Semantic-UI is probably the best CSS solution for now. It is much more flexible and elegant than Bootstrap which I had used for a long time before meeting this masterpiece.
Recently React is becoming a game changer for this framework. No wonder since many of us know how good is it when used as a React UI library.
[Semantic UI for React trends](https://trends.google.com/trends/explore?q=semantic%20ui%20react)

**Comment by levithomason** [comment]
Do we have an agenda for this meet?

**Comment by msmiller** [comment]
I hope SUI can keep going. It's the best I've found (especially for Rails with all it's dependencies)
 and my default framework for all new projects.

This week I decided to play around with Rails 5.2 for a new codebase I'm tinkering with. SUI is the only one that came up clean the first time without needing any major tweaks or configuration changes. Semantic wouldn't fire up at all - none of the JS worked. And Bootstrap required a bunch of other jQuery gems and stuff to get working - plus it's on the fat side, and doesn't look as clean as SUI.

SUI really is a hidden gem. A shame more people don't know about it.

**Comment by menthol** [comment]
Please record the Hangout session on youtube. Sure many of us want to know what was said during the session.

**Comment by hammy2899** [comment]
@menthol that might not be a bad idea. 

Gitlab live stream a lot of their meetings/discussions. [Gitlab's YouTube](https://www.youtube.com/channel/UCnMGQ8QHMAnVIsI3xJrihhg)

**Comment by GuacheSuede** [comment]
@jlukic will you post the hangout link here ?

**Comment by jlukic** [comment]
I'm not sure I have an agenda, does anyone have suggestions. I'll look into recording as well

**Comment by levithomason** [comment]
I know we'll discuss the long-term and grand vision for SUI.  I would like to address some of the shorter term goals and current issues.  Being the React integration author/maintainer, there are things I'd like to do there in terms of themes and styling that I can't reconcile entirely with SUI.

Jack, this is similar to what we've talked about before with abstracting styles into a central location so that we can generate styles for all integrations.  There are some other implications I'd like to consider here as well, such as removing child based selectors and isolating styles between components (e.g. a label in a menu item wouldn't change its style).

I think my focus is primarily on where we are now and how we take the next immediate step.  Unfortunately, I've become quite ill and I've completely lost my voice 😕 There's a good chance I won't be able to join.

If you, and the others, could address a feasible path to moving forward with what and who we have right now, that would address what I'm most focused on.  The other option is to fork Semantic UI React into a new effort that handles these things, but I think it would be a shame to split what little forces we have already.

**Comment by GuacheSuede** [comment]
@levithomason you can just listen in i would hope, your insights would be valueble

**Comment by levithomason** [comment]
I will do my best 🙇

**Comment by DanAtShenTech** [comment]
What is the link for the hangout?

**Comment by jlukic** [comment]
https://hangouts.google.com/hangouts/_/kxc753flhfdfxmz75ifrxymrbye

**Comment by nuthingman** [comment]
How was the meeting? Any decision regarding the future of Semantic-UI?

**Comment by levithomason** [comment]
I believe the results were recorded on YouTube.   It was at the least broadcast live.  The results were promising.  Jack took some good notes and recorded some actionable items.  The high-level picture is that steps will be taken to reduce the number of open issues and keep progress going while the larger funding issue is tackled.

**Comment by GuacheSuede** [comment]
@levithomason Do you have a link ?

**Comment by levithomason** [comment]
I don't, I just saw the `live` indicator in the upper right of the call.

**Comment by Laerte** [comment]
@jlukic Can you provide the link of recorded hangout?

**Comment by arjarn** [comment]
No feedback about meeting ?
Short summary will be great ;-)

**Comment by dbendaou** [comment]
Any news ?

**Comment by GuacheSuede** [comment]
From Where I stand, i have reason to beleive Semantic UI is dead.
Jack has not replied my many emails or gitter messages. 
I feel that nothing was highlighted nor any plan was created during the Hangouts chat ( I was there ) and 
** Jack has not come out with any funding plan. in fact he refused due to his "idiosyncrasies" **
Would anyone be interested in a community fork ?

**Comment by dbendaou** [comment]
As it as already been told, there seem to be people willing to support SUI, maybe @jlukic or @levithomason considers adding them to the team...?

**Comment by hammy2899** [comment]
From what I gathered Jack made some notes from the hangout and did start making some sort of a plan and he did agree to add the people who where in the hangout to the semantic team to help maintain it.

I think we need to wait and see what Jack says before we make a fork because we don't exactly know why he isn't responding, anything could have happened.

Lets wait another week or so before we start planning a community fork.

**Comment by dannyBies** [comment]
I would be up for helping to maintain the project. I and probably many others would love to help but at the moment it feels like not a lot of progress is being made and there is no good way to help the Semantic UI community.

**Comment by brodybits** [comment]
In general Semantic-UI seems to both offer unique features and try to do other things that other frameworks such as Bootstrap, Bulma, Foundation, Polymer, etc. already specialize in doing better. I would be much happier to see Semantic-UI focus the precious resources of its community on the unique value that it can provide.

**Comment by GuacheSuede** [comment]
@hammy2899 **Btw, it has been 2.5 weeks of no activity since the chat excluding a Month of no updates other than the one time we pushed Jack to get the Github Issue out.**

**Comment by levithomason** [comment]
For clarity, SUI development is not dead.  It is, however, quite slow.  You have to realize what it is like to have a repo with 40k stars.  Jack had somewhere around 8k Github notifications on the call we had.  It is unrealistic to think he can address every request personally.  Please realize, we both have full-time jobs that take first priority of our development time.

It is also safe to say that SUI is not currently undergoing drastic changes or improvements.  The React port is approaching a similar position, though, continues to iterate.  In the background, I can say I have some experiments going that may influence the future of SUI's theming system.

I would forgo public opinion in favor of data regarding progress, though.  Here's the last month's worth of data on development in Semantic Org.  I've also shown user adoption continues to increase for both projects.

# Repo Pulse

## Semantic UI

https://github.com/Semantic-Org/Semantic-UI/pulse

![image](https://user-images.githubusercontent.com/5067638/36335745-03ad7672-1337-11e8-8e3e-9c3d7832950c.png)

## Semantic UI React

FWIW, the React port which I manage has the lion share of development and progress these days:

https://github.com/Semantic-Org/Semantic-UI-React/pulse

![image](https://user-images.githubusercontent.com/5067638/36335738-fa13247c-1336-11e8-9665-5a051713da01.png)

# NPM Downloads

Users aren't slowing down on package installs either.  

## Semantic UI

https://npm-stat.com/charts.html?package=semantic-ui&from=2015-01-01&to=2018-01-31

![image](https://user-images.githubusercontent.com/5067638/36335828-b509b584-1337-11e8-9d32-492418612d0d.png)

## Semantic UI React

https://npm-stat.com/charts.html?package=semantic-ui-react&from=2016-09-01&to=2018-01-31

![image](https://user-images.githubusercontent.com/5067638/36335855-f4128dc8-1337-11e8-8b0a-86b40ef892d4.png)

Not sure why this tool reports 173k, we're actually pushing 200k downloads a month.

![image](https://user-images.githubusercontent.com/5067638/36335882-344483c4-1338-11e8-8dae-97de7d208835.png)

***

It is easy to think of software like other paid products and services.  However, OSS is free and it is often borne out of the passion and sacrifice of a small few and offered to the world for the sake of doing exactly that.  Please try to remember that when setting expectations and making requests.

I will leave with a call to action.  I started developing on SUI in 2015 out of my own needs.  What has astounded me is the proportion of users asking for more and not giving back.  As a maintainer, I'm always hoping someone will jump in and join forces in the mission, making a better web framework.  I'm beyond grateful for the few that have, but I dream of a day when initiative, contribution, and perseverance outpace the barrage of requests.  Short of personal responsibility and action, we'll simply have to wait for Jack to figure how to arrange for indefinite payment for himself and others to bring improvements full time.  So, roll up your sleeves, fork repos, submit PRs, assemble issue triage teams, engineer solutions, and build the thing you are waiting for.  The world is before you and it is yours to take.  We're here.

**Comment by levithomason** [comment]
Reading back, that sounds a little harsh and whiny 😸 Totally not my intent, just trying to shed a much reality as I can on the topic.

**Comment by levithomason** [comment]
Agreed.  I'll do my best to answer from my perspective.  This also prompts me to discuss what I think is an upcoming split between SUI and SUIR.

>What is the value or unique quality (as `@brodybits` mentioned) that SUI provides? What distinguishes it from the other UI frameworks out there?

For me, it is the semantic design language and it's thorough application to components.  Explicitly defining component types, content, states, and variations and applying those in matrix fashion across all components makes SUI one of the most feature complete frameworks there is.  In the React port, I can often change the name of a component only, leaving all props and values as-is, and the get a brand new experience that still works.  This is because all components share this common language and apply it effectively.

Second, theming.  Although the current process is a royal PITA, the power of SUI's theming capabilities is unparalleled.  This is why I hope to help reign it in someday with a more user-friendly and shareable theming experience.

These core values will follow me everywhere I go.

>What is SUI's purpose?

**SUI**
Jack has answered this better than I can.  In an attempt to summarize, Jack's lifelong pursuit is something of a W3C for UI.  A standard for the language of the web components.  SUI itself is something of a paramount example implementation of what that might look like.

**SUIR**
I created this strictly for business purposes.  I needed to build apps with a highly themeable framework that included as many components and variations as possible to avoid reinventing the wheel.  This is also why we continue adding "addon" components to the library that don't exist in SUI core.  It's purpose is simply to be the best React component library available.  Best to me means tests, docs, maintenance, and a plethora of components and capabilties.

>What is the plan henceforth?

**SUI**
We're pending Jack's response here for SUI, but he indicated he's still looking for ways to pay for development.

**SUIR**
My aspirations and plans are a little different than Jack's.  I plan to continue tackling immediate business needs for component libraries.  In fact, I'm starting a new job at a prominent employer and will be tasked with doing exactly that on a global enterprise level very soon.  As such, I will either be making drastic updates to SUIR to support this mission or (more than likely) I'll drop SUI's CSS and ship a new component library heavily inspired by SUIR.

>How do we accomplish SUI's purpose?

**SUI**
Again, we're awaiting Jack's notes and direction for SUI core.

**SUIR**
Our purpose is to design the most friendly and powerful component API there is.  We have a very clear README section outlining exactly how you can help.  We have specific issue labels identifying issues needing comment/feedback, needing engineering, needing general help, and good first issues for those getting started.  We also have an extensive CONTRIBUTING.md detailing how to create components and use our stack.

>How do maintain a UI framework in a world that changes?

This is the key question for me!  How do we not constantly rewrite so much code everytime technology moves on?  How do we share components and styles between static pages, jQuery sites, React apps, native mobile apps, and the next JS framework to overtake the community?  How do we create a framework that is simple to theme and can be used by any business or enterprise?

These questions are exactly what has pushed me to my most recent conclusions and prototypes.  Thank goodness, they are also the reason for my new job opportunity where I'll get to build them out.

I believe the answer is to abstract and generalize as much of the duplicated work as possible.

1. Shared styles.  SUI CSS is powerful but it is also limiting, IMO.  It is written in LESS and only works in a browser.  Also the pervasive use of child selectors and important make it difficult to integrate elsewhere.  Following the SUI design language, create a framework agnostic style framework that can be used anywhere and compile down to any format.  JS is the best candidate for this in my mind.

2. Shared state.  SUI and SUIR have duplicated control flow for the Dropdown component with thousands of lines of JS each.  This is a tragedy.  Once Vue is the new hotness, we'll do it a 3rd time.  That is insanity.  Many modules are well represented as finite state machines.  I want to further explore this and how they can be mapped to frameworks.

>As a community or as maintainers or as the creator, when those questions are answered, a plan can follow. Once the plan is communicated, the few and precious resources that others can offer can be focused and used wisely.

Exactly.  I can't speak for SUI core as that is Jack's alone.  Part of my new role may include building this framework as I'm describing, evangelizing it, and building community.  I can't guarantee that nor that it will be soon if it does happen, but it is at least the meeting of minds we have at the moment.

**Comment by dannyBies** [comment]
Thank you for your detailed answer Levi.

> I will leave with a call to action. I started developing on SUI in 2015 out of my own needs. What has astounded me is the proportion of users asking for more and not giving back. As a maintainer, I'm always hoping someone will jump in and join forces in the mission, making a better web framework. I'm beyond grateful for the few that have, but I dream of a day when initiative, contribution, and perseverance outpace the barrage of requests. Short of personal responsibility and action, we'll simply have to wait for Jack to figure how to arrange for indefinite payment for himself and others to bring improvements full time. So, roll up your sleeves, fork repos, submit PRs, assemble issue triage teams, engineer solutions, and build the thing you are waiting for. The world is before you and it is yours to take. We're here.

Back in July I submitted my first pull request to Semantic UI (https://github.com/Semantic-Org/Semantic-UI/pull/5507). At my job we were using my forked version of SUI until it was fixed in 2.2.14, for which me and my colleagues are really grateful. 

In the period 5th of July - 19th of December nothing happened with my pull-request. On the 19th I started having email contact with Jack to see if my fix could be merged and if there was any way to help him out. This ended up in Jack promising a resolution by feb 1 which resulted in 2.2.14.

I understand how difficult it is to have a fulltime job and maintaining SUI. But at this moment as a developer I don't have a good reason to be active in the SUI community. I see pull requests being made every week that will either not be merged at all or it will take months before they will be added in a new version. 

I would love to be active here, but currently it feels like a waste of time.

**Comment by hammy2899** [comment]
@dannyBies I agree we do need some more people who can approve PR. Jack discussed this in the hangout of possibly making PR need 2 people to approve them before it can be merged and I think this would work really well but SUI is lacking maintainers with said power.

**Comment by levithomason** [comment]
I also agree.  I don't use the majority of SUI, just the styles, as the React port includes all of it's own JS and renders it's own HTML.

That said if I were in this position with SUI, I would consider forking and maintaining that.  If I truly thought I could do a better service, I would do it.  When someone else came along (possibly Jack) and resumed providing a better service than I was able to, then I'd gladly merge and join forces.

This is actually what happened in the Node.js community with io.js.  There were some differences and disagreements, so io.js forked from Node.js and took off.  After considerable progress, everything was resolved and io.js merged back with Node.js.

I guess I'm giving another call to action.  Don't stop at submitting a PR.  Just solve the problem in front of you.  If that problem is PRs not getting merged, figure out a way to merge them.  Creating a fork is one way.  If there is not a clear plan for making enhancements, fork the project, create a plan and invite everyone to join you.  The world is truly here for your taking.

Consider all the active people in this thread who feel as you do.  It seems to me they just need a leader to pull it together.  Why not be that leader?

**Comment by levithomason** [comment]
I know one hurdle to merging PRs is getting thorough and proper reviews.  I wonder if we could set up some kind of process or tool to merge PRs based on serveral reviewer approvals.

I do have this problem at SUIR as well.  We've added a number of contributors to help but it would be great to leverage more of the community for this process.

I imagine having a core group of folks who can skim and sign off on code quality and then open to the community for reviewing and approving the feature/fix itself.

**Comment by dannyBies** [comment]
Forking should be a last resort as splitting an already small community will most likely end in disaster. I do think the only solutions are to fork or to add a group of people chosen by @jlukic to help maintain the project.

I say we wait until the 25th to see if we can find a solution and if not, we fork with the idea that we can merge back to the main repo when we do find a viable solution.

I do not have a lot of time to be the owner of the fork, so if someone else has the time and wants to do it (with me) comment here. If not I will fork on the 26th.

**Comment by hammy2899** [comment]
@dannyBies I agree and would be happy to help you maintain the fork or be the owner if you don't have the time.

**Comment by levithomason** [comment]
I would start with small bug fixes only and ship a new npm package to users can install it.  This should make it easier to merge back to the main repo later.  Also, staying current with the main repo will be important.

My hope is that a fork would produce a team of maintainers and prove that bug fixes can be shipped reliably.  Perhaps this will be enough of a signal to allow those maintainers to merge to master on the main repo.

**Comment by msmiller** [comment]
Bulma is nice, but it's not as complete a solution as SUI. You need to add in form validators and something like Masonry and a bunch of other stuff to match the overall functionality. Bulma is pretty strict about "mobile first" too - building a robust Rails app is kind of a pain when you have to keep trying to work around the assumption it makes.

**Comment by arjarn** [comment]
I'm really sad to read this. In my mind SUI (without "R" ;) ) is a great and unique framework..
Multiples bricks we can use to build various site easily. I never see this approach before and like it.
Idea about W3C UI is very seducing. 
Personnally, I'm needing framework without complexity of SUIR (great for professionnal using) cause I'm doing stuff as a hobby (but it's not incompatible with love good job with nice software tool).

I hope so @jlukic communicate about it and why not explain plan. It seem motivated people here who coul help him to realize. Without that, SUI dead slowly...

**Comment by jlukic** [comment]
I spent my holiday getting out `2.3.0`, you can view the full notes [here](https://semantic-ui.com/introduction/new.html), flexbox modals, font awesome 5, some other stuff.

I think the current speed of releases is healthy, although its clearly not the same as having dedicated full time work on SUI. 

You can watch the Semantic UI [discussion here](https://www.youtube.com/watch?v=cZ9lMzv0kTo). 

The outline of general project modifications were
-- Setup CSS Regression testing
-- Setup Stalebot <https://github.com/probot/stale> to remove old conversations
-- Add new issue template
-- Consider bug hunt day

And for user contributions
-- Have system for code review for trusted contributors for PRs
-- Have volunteers to help maintain the discussion boards

**Comment by dannyBies** [comment]
Thank you for the new version and the information. This sounds really good to me, if we can work on these points I don't see a reason to fork.

**Comment by levithomason** [comment]
@jlukic see stalebot setup at #6182.  Same setup as SUIR has been using.  I will merge it soon and add the bot.

**Comment by arjarn** [comment]
Nice video ! 
I understand @jlukic  point of view with core and modules as calendar, slider...
But perhaps community could developp "independant" modules working with SUI. So if you're needing image slider or other, you could find and use it. So core can keep slim for summer ^^
Seriously, I continue thinking community components is real advantage for dev : no need using obscure 3rd party components, no nightmare for maintenance.

**Comment by w96k** [comment]
@levithomason @jlukic What do you think about the opportunity to implement semantic-ui using styled components (for 3.0 maybe)? IMO this way we can merge sui and suir together and focus on maintaining one repo. With styled components we don't even need to use less, js have better data structures (simple obj, for example LESS doesn't have a nesting like objects do in js, so many vars have repeated words in their names) and styled components have a good solution for theming <ThemeProvider>. You don't need to separate view and functional and can use fully-functional components pretty easily from the box. Styled components can be easily rendered on server & used on mobile with react-native.

**Comment by levithomason** [comment]
I agree 99.9% on all points.  I've written about this for a couple years now, as in https://github.com/Semantic-Org/Semantic-UI-React/issues/1009.  It would also enable sharing the same styles between the web (jQuery, React, etc), native mobile apps (React Native), and desktop (electron) apps.  I've spoken all too much about the benefits in many issues over the years.  I also have prototype of this on my machine now.  Another contributor.

Jack and I have different opinions on this and he has good counterpoints as well.  I still believe CSS in JS is not only the best solution but the future direction of the web.

The 0.1% difference I have with your comment would be the implementation library. I wouldn't use styled components, probably `fela` or some other.

As I've eluded to recently in a few issues, I may have a fulltime opportunity to make this happen in 2018.  Really hoping it pans out.

**Comment by sindzicat** [comment]
Hello all!

What about crowdsourcing? SUI owners could create Trello board or something like this. They create small tasks for SUI community, and folks try to do these tasks. My idea is that owners have some free time, but their time isn't enough to write code. But their time may be enough to organize development.

I'm a mechanical engineer. There is a technical director on my firm, and he know what machine should be developed. He give me small tasks, so that I know what exactly and how I should to do. I don't know all moments about my machine. Every time I do only not big tasks, not more.

In my opinion tasks must be small. And person, who want and can help SUI, can do such task in one or a few days. As for SUI owners it should be easy for them to check the result of tasks made. This check should not spent much their time, and it should be easier for them to find free time.

**Comment by mvrhov** [comment]
Having css and js and html intermingled means that I'm forced to render with javascript. this means that either I have to render client side which means that there will be delay. If I want to render this serverside then I'm forced to also install nodejs along with the technology I use for my backend. So That a big 👎 👎

**Comment by w96k** [comment]
@mvrhov I don't see much problem here, I saw how in go/php people using embed js-engine and rendering all the stuff they needed just fine. If you are using semantic ui you anyway need to install node.js for theming, because gulp & less are js written libs. I got your point and I think cross language boundaries is a big problem not only in react ecosystem, but in computer science in general.

**Comment by khornberg** [comment]
@jlukic Thank you for your response and time to answer.

At this point I think this issue is resolved. Semantic UI is not dead. There is a community that wants to keep it going.

I think it would be helpful to create an RFC repo to discuss future direction of the project and the planning of the implementations of the decisions we make.

**Comment by mikelax** [comment]
Having just starting to explore and use semantic UI and semantic react I have to say these packages are a real joy to use. I was going to offer a few thoughts that might help prevent forking of the project, but there seems to be a flurry of activity over the past two days. 

@jlukic I think your post outlining the roadmap and proposed changes goes a long way to addressing some of the community concerns !! 🥇 

I know the community is there to support the project, that is evidenced by the number of issues and PRs. The issue (in my small opinion) is that this project is too big for any single person to fully manage. This would be try for anyone full-time, let alone on a part-time basis,. Getting time from the community to help with things like issue triage, PR review, will go a long way. 

Here are my couple of thoughts (most of which you already outlined). Take them for what they are, but it already appears things are moving in the right direction 📈 💯 

* Create issue template to doubly signal to users that usage and "how-to" questions should be directed to Stack Overflow. 
* Triage existing issue list and close invalid issues, mention to open question on SO if still needed
* Get a group of power users that can lend time offering to do PR reviews, and comments. 
  * Quick win - do a small point release and try to include a bunch of smaller PRs, bug fixes, to test the waters and begin to clear out the PR backlog. 
* Create a Roadmap.md file or add section in main README to list upcoming features

**Comment by brodybits** [comment]
-1 for Stack Overflow, I do not think it provides consistent support. I think a help forum is badly needed for this project.

**Comment by khornberg** [comment]
http://forums.semantic-ui.com/

**Comment by mikelax** [comment]
@brodybits @khornberg whether the "help and support" is done through SO or custom forum software I don't really mind either way. I only mention to not have those questions clutter the Github Issues. It's the same goal. 

The issue with running custom forum software is that you have to then recreate the dev community that already exists on SO. ie. there are more [questions tagged semantic-ui](https://stackoverflow.com/questions/tagged/semantic-ui) on SO than in the forums. The other big benefit of SO is their SEO is killer, if the questions get answered then people with the same question *will* find those existing questions, that's hard to beat.

**Comment by hammy2899** [comment]
I want to add to that by saying most people don't even know about the current forums (which was just demonstrated) and most devs go straight to SO for any programming related issues anyway.

**Comment by kielcoding** [comment]
@w96k it’s not the main issue here, but I just want to give my opinion to the css in js thing. First your argument that you already usually have js for developing semantic is invalid because development environment != production environment in general.
Second it might sound good to at first to solve css issues by the help of ja but in my opinion you increase the complexity and add additional potential problems. So instead of struggle with css you do it with js. There are a lot of people who prefer the first I guess :) if you want to extract the css part from SUI than I want say the best way would be a less/sass only repository. But that what only be a benefit for external projects such as SUIR not SUI in general.

**Comment by hammy2899** [comment]
@kielcoding There is already CSS and LESS only repos.

https://github.com/Semantic-Org/Semantic-UI-CSS
https://github.com/Semantic-Org/Semantic-UI-LESS

**Comment by kielcoding** [comment]
@hammy2899 Yes, I know, but they are only _extraction_, not dependencies or any other kind of reusable code/abstraction.

**Comment by brodybits** [comment]
@jlukic my understanding from your talk is that you had some issues with Semantic UI together with web components. Is this correct and can you explain further?

**Comment by jlukic** [comment]
@brodybits  Pretty good overview [in this article](https://dmitriid.com/blog/2017/03/the-broken-promise-of-web-components/).

**Comment by brodybits** [comment]
> @brodybits Pretty good overview in [this article](https://dmitriid.com/blog/2017/03/the-broken-promise-of-web-components/).

... now with a link to [an excellent response](https://robdodson.me/regarding-the-broken-promise-of-web-components/), more details in <https://github.com/Semantic-Org/Semantic-UI/issues/4963#issuecomment-369732377>.

I start to wonder if it would make more sense to get the JavaScript part out of this project?

**Comment by slaklad** [comment]
Are you @levithomason  and @jlukic  still on a regular job? I thought the project was paid off with ads and donations.

**Comment by levithomason** [comment]
Unfortunately, ads and donations cannot support even one junior developer's salary.  I'm a full-time employee, husband, dad, and landlord doing my best to support SUI in my "free" time 😅

**Comment by prudho** [comment]
I'm sorry to pull up this subject again, but is there someone to maintain this repository alive ? I mean, pull requests aren't reviewed, issues are stacking without any response or even closed (just take a look at duplicates and no semantic related issues)...

There is some contributors who are trying to help, but without any reaction from maintainers this is very frustrating. Some bugs are critical and should be fixed in a hotfix without waiting the next big milestone.

**Comment by w96k** [comment]
Yeah, would be nice if someone will have a control over the repo, maybe @hammy2899 can review and merge stuff?

**Comment by hammy2899** [comment]
I would be happy to help out by merging some PR's but it's down to @jlukic and I want to respect his work. I don't want to feel pushy and as if I am taking over his project.

**Comment by jlukic** [comment]
I think we're covered this topic adequately so I'm closing discusion. I recommend those who discover this thread to watch the video and check my comments above, specifically related to the development history of the project.

@slaklad The donations/ads is about one tenth of a starting devs salary. So its not something that can warrant full time development without eating into savings dramatically.

**Comment by morganbarrett** [comment]
I'm starting active development on this non jQuery fork: https://github.com/morganbarrett/Vanilla-Semantic-UI
